### PR TITLE
[EPT-1709] API layer to interact with uiapi/page

### PR DESCRIPTION
**Pull request by shanthru**
API layer to interact with uiapi/page.
http.js is copied from font-management-frontend and email-composer

### CODE REVIEW

File: src/core/api.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export function fetchPages() {
+    return http.get("/uiapi/page");
+}
+
+export function createPage(data) {
+    return http.post("/uiapi/page", data);
+}
+
+export function updatePage({ pageId, data }) {
+    return http.patch(`/uiapi/page?id=${pageId}`, data);
```
Comment (lindsaycooper) [inline src/core/api.js:12]: @rvijendran-sailthru FYI the frontend is using PATCH for now. This is changeable.

File: src/core/api.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export function deletePage(pageId) {
+    return http.delete(`/uiapi/page?id=${pageId}`);
```
Comment (lindsaycooper) [inline src/core/api.js:16]: question: Based on the comment above, should this code be written this way?
```suggestion
    return http.delete('uiapi/page', { id: pageId });
```

**Review by lindsaycooper — APPROVED** [review]

**Review by lindsaycooper — COMMENTED** [review]
PR #1 seems like a good place to update the README

File: src/core/http.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+/**
+ * Adds shortcut methods to http to make usage less verbose
+ *
+ * Query params should be passed as an object instead manually
+ * concatenating the url str
+ *
+ * Examples:
+ *  http.get('uiapi/templates', { version: "v1" });
+ *      => GET https://my.sailthru.com/uiapi/templates?version=v1
+ *  http.delete('uiapi/templates/123');
```
[the remark this answers is not in the document]
Comment (shanthru) [reply src/core/http.js:58]: This comment was originally written as an example in email-composer that uses this structure for templates.
It was also copied to font-management with the same example although the api actually uses `?id=${fontId}`

Update the example, the code or wait until the backend?

File: src/core/api.js (hunk 1 above)
Comment (shanthru) [reply src/core/api.js:12]: This is subject to change based on decisions for the backend.

**Comment by shanthru** [comment]
> 

Updated README :) Needs improvements but has a starting point

File: src/core/http.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lindsaycooper) [reply src/core/http.js:58]: non-blocking: It seems like this file is adding value but also adding some confusion too (we aren't using some of the features). Wondering if its worth the effort to try to simplify it to our specific use case here. And if it's simple enough, could the logic be moved in api.js

cc @jdrury-sailthru

[the remark this answers is not in the document]
Comment (lindsaycooper) [reply src/core/http.js:58]: Examples of this:
https://github.com/sailthru/email-composer/blob/main/src/features/Experiences/experiencesApi.js
https://github.com/sailthru/email-composer/blob/main/src/features/Fonts/fontApi.js

File: src/core/api.js (hunk 1 above)
Comment (jdrury-sailthru) [reply src/core/api.js:12]: there may also be a way to pass params via function args

**Review by jdrury-sailthru — APPROVED** [review]
LGTM but i have some misgivings about using http (i like that util, but I wrote it)

File: src/core/api.js (hunk 1 above)
Comment (jdrury-sailthru) [reply src/core/api.js:12]: there is but it is the 4th function arg

```suggestion
    return http.patch(`/uiapi/page`, data, , { id: pageId } );
```

Comment (jdrury-sailthru) [reply src/core/api.js:12]: (not sure if javascript supports empty args like in that example)

**Pull request merged**